packages/fxa-auth-server/scripts/move-customers-to-new-plan-v2/move-customers-to-new-plan-v2.ts

```diff
@@ -0,0 +1,273 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Stripe from 'stripe';
+import { writeFile } from 'fs/promises';
+import PQueue from 'p-queue';
+import { PayPalHelper } from '../../lib/payments/paypal';
+
+import { ACTIVE_SUBSCRIPTION_STATUSES } from 'fxa-shared/subscriptions/stripe';
+import { RefundType } from '@fxa/payments/paypal';
+
+export class CustomerPlanMover {
+  private stripeQueue: PQueue;
+
+  constructor(
+    private sourcePlanId: string,
+    private destinationPlanId: string,
+    private excludeCustomersHavingPriceIds: string[],
+    private outputFile: string,
+    private stripe: Stripe,
+    public dryRun: boolean,
+    rateLimit: number,
+    private proratedRefundRate: number | null,
+    private coupon: string | null,
+    private paypalHelper: PayPalHelper,
+  ) {
+    if (proratedRefundRate !== null && proratedRefundRate <= 0) {
+      throw new Error("proratedRefundRate must be greater than zero");
+    }
+
+    this.stripeQueue = new PQueue({
+      intervalCap: rateLimit,
+      interval: 1000, // Stripe measures it's rate limit per second
+    });
+  }
+
+  async convert() {
+    await this.writeReportHeader();
+
+    await this.stripe.subscriptions.list({
+      price: this.sourcePlanId,
+      limit: 100,
```

Comment:
Can this be raised to ~~1000~~ the rate limit? Since this is likely going to be used to batch-process a large amount of users, I'd love to minimize the number of api calls we need to make.